CodeMaid/Helpers/SettingMonitor.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace SteveCadwallader.CodeMaid.Helpers
+{
+    public sealed class SettingMonitor<TSetting> where TSetting : ApplicationSettingsBase
+    {
+        private readonly Dictionary<string[], Monitor> _monitors = new Dictionary<string[], Monitor>(new StringArrayComparor());
+
+        public SettingMonitor(TSetting settings)
+        {
+            Settings = settings;
+            Settings.SettingsSaving += (s, e) => NotifySettingsChanged();
+        }
+
+        public TSetting Settings { get; }
+
+        internal void NotifySettingsChanged()
+        {
+            foreach (var item in _monitors)
+            {
+                var monitor = item.Value;
+                var oldValues = monitor.LastValues;
+                var newValues = FindValues(item.Key);
+                if (!Enumerable.SequenceEqual(oldValues, newValues))
+                {
+                    monitor.LastValues = newValues;
+                    monitor.Callback.Invoke(newValues);
+                }
+            }
+        }
+
+        private object[] FindValues(string[] settings) => Array.ConvertAll(settings, key => Settings[key]);
+
+        public void Watch<TValue>(Expression<Func<TSetting, TValue>> setting, Action<TValue> changedCallback)
+        {
+            var settingName = (setting.Body as MemberExpression).Member.Name;
+            Watch<TValue>(new[] { settingName }, values => changedCallback.Invoke(values[0]));
+        }
+
+        public void Watch<TValue>(string[] settings, Action<TValue[]> changedCallback)
```

Comment:
As what I typed in another comment, for me, it's better to add this method, it can simplify call code and prevent a duplicated called issue. I think making call code simpler takes precedence over making implementation complexer.